contracts/margin/external/BucketLender/ProductionBucketLender.sol

```diff
@@ -0,0 +1,103 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+import { Math } from "openzeppelin-solidity/contracts/math/Math.sol";
+import { BucketLender } from "./BucketLender.sol";
+import { MarginHelper } from "../../external/lib/MarginHelper.sol";
+import { MarginCommon } from "../../impl/MarginCommon.sol";
+
+
+/**
+ * @title ProductionBucketLender
+ * @author dYdX
+ *
+ * Extension of BucketLender that delays the force-recovery time
+ */
+contract ProductionBucketLender is BucketLender
+{
+    // ============ State Variables ============
+
+    // number of seconds after position has closed that must be waited before force-recovering
+    uint32 RECOVERY_DELAY;
+
+    // ============ Constructor ============
+
+    constructor(
+        address margin,
+        bytes32 positionId,
+        address heldToken,
+        address owedToken,
+        uint32[7] parameters,
+        address[] trustedMarginCallers,
+        address[] trustedWithdrawers,
+        uint32 recoveryDelay
+    )
+        public
+        BucketLender(
+            margin,
+            positionId,
+            heldToken,
+            owedToken,
+            parameters,
+            trustedMarginCallers,
+            trustedWithdrawers
+        )
+    {
+        RECOVERY_DELAY = recoveryDelay;
+    }
+
+    // ============ Margin-Only State-Changing Functions ============
+
+    /**
+     * Function a contract must implement in order to let other addresses call
+     * forceRecoverCollateral().
```

Comment:
make it clear that this is overriding `BucketLender`'s `forceRecoverCollateralOnBehalfOf`